src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -489,7 +490,13 @@ double BaseCouplingScheme::getNextTimeStepMaxSize() const
   }
 
   if (hasTimeWindowSize()) {
-    return _timeWindowStartTime + _timeWindowSize - _time;
+    double maxDt = getWindowStartTime() + _timeWindowSize - _time;
+    if (math::equals(_maxTime, UNDEFINED_MAX_TIME)) {
+      return maxDt;
+    } else {
+      double leftover = _maxTime - _time;
+      return std::min(maxDt, leftover);
```

Comment:
Do we really want to forward here an arbitrary small dt? I would say that the current behavior is numerically still more robust. Not sure that this PR is the right place to add it into the library.

Reply:
With participant-first methods and adaptive time-stepping (e.g. in OpenFOAM) this can still lead to issues. In the adapter, we set the end time to infinity such that preCICE steers it. Hence, the solver is not aware of the maxTime limit and this return value can lead o arbitrary small dt causing numerical issues then.

Reply:
I just realized that this only applies to `hasTimeWindowSize()`. What happens to the second participant in a `participant-first` time-stepping. Do we then perform different time-step length for different participants?

Reply:
For `participant-first`, the `getNextTimeStepMaxSize()` is maxTime if provided and inf otherwise.